pkg/api/v1alpha1/nutanix_types.go

```diff
@@ -0,0 +1,30 @@
+package v1alpha1
+
+// NutanixIdentifierType is an enumeration of different resource identifier types.
+type NutanixIdentifierType string
+
+const (
+	// NutanixIdentifierUUID is a resource identifier identifying the object by UUID.
+	NutanixIdentifierUUID NutanixIdentifierType = "uuid"
+
+	// NutanixIdentifierName is a resource identifier identifying the object by Name.
+	NutanixIdentifierName NutanixIdentifierType = "name"
+)
+
+// NutanixResourceIdentifier holds the identity of a Nutanix Prism resource (cluster, image, subnet, etc.)
+//
+// +union
+type NutanixResourceIdentifier struct {
```

Comment:
Could we move the struct definition to the place of consumption? Since they are used inside `NutanixMachineConfig`.